### PR TITLE
Add mitigation for failed authentication sessions

### PR DESCRIPTION
**Pull request by ollityrkko**

### CODE REVIEW

File: src/iqm/iqm_client/authentication.py [hunk 1]
```diff
@@ -48,21 +48,24 @@ class TokenManager:
...
     @staticmethod
-    def time_left_seconds(token: str) -> int:
+    def time_left_seconds(token: Optional[str]) -> int:
         """Check how much time is left until the token expires.
 
         Returns:
             Time left on token in seconds.
         """
-        if not token:
+        if not token or not isinstance(token, str):
```
Comment (kukushechkin) [inline src/iqm/iqm_client/authentication.py:57]: Function signature says `Optional[str]`, but here we check if it is `str`. What else could it be? Method signature should be updated as well then if it is expected.

File: src/iqm/iqm_client/authentication.py [hunk 2]
```diff
@@ -48,21 +48,24 @@ class TokenManager:
...
             return 0
         parts = token.split('.', 2)
         if len(parts) != 3:
             return 0
         # Add padding to adjust body length to a multiple of 4 chars as required by base64 decoding
-        body = parts[1] + ('=' * (-len(parts[1]) % 4))
-        exp_time = int(json.loads(b64decode(body)).get('exp', '0'))
-        return max(0, exp_time - int(time.time()))
+        try:
```
Comment (kukushechkin) [inline src/iqm/iqm_client/authentication.py:63]: Is there a reason not to use `jwt` dependency here?

Comment (ollityrkko) [reply src/iqm/iqm_client/authentication.py:63]: `jwt` dependency would be an overkill for extracting just one value with no validation.

File: src/iqm/iqm_client/authentication.py (hunk 1 above)
Comment (ollityrkko) [reply src/iqm/iqm_client/authentication.py:57]: Changed the signature to `Any` and improved token validation.

Comment (freetonik) [reply src/iqm/iqm_client/authentication.py:57]: Can it be anything other than `str` and `null`?

Comment (ollityrkko) [reply src/iqm/iqm_client/authentication.py:57]: It should never be anything else than `str` or `None`, but `time_left_seconds` is also used to validate a token, so it should be able to handle any given value without crashing.
Especially, `TokenClient` will try to validate anything the server sends as "refresh_token", so better be safe than sorry.

**Review by freetonik — APPROVED** [review]

**Pull request merged**